File: templates/problems/19_acm_io/04_matrix_input/typescript/solution.ts
/**
 * Read an n x m matrix and return its transpose (m x n) as a string.
 * Time: O(n*m), Space: O(n*m).
 */
export function solveMatrixInput(inputData: string): string {
    const tokens = inputData.split(/\s+/).filter(t => t.length > 0);
    let idx = 0;
    const n = parseInt(tokens[idx++], 10);
    const m = parseInt(tokens[idx++], 10);

    const matrix: number[][] = [];
    for (let i = 0; i < n; i++) {
        const row: number[] = [];
        for (let j = 0; j < m; j++) {
            row.push(parseInt(tokens[idx++], 10));
        }
        matrix.push(row);
    }

    const outLines: string[] = [];
    for (let j = 0; j < m; j++) {
        const col = matrix.map(row => String(row[j]));
        outLines.push(col.join(" "));
    }
    return outLines.join("\n") + "\n";
}

File: templates/problems/19_acm_io/04_matrix_input/python/solution.py
def solve_matrix_input(input_data: str) -> str:
    """Read an n x m matrix and return its transpose (m x n) as a string.

    Time: O(n*m), Space: O(n*m).
    """
    tokens = input_data.split()
    idx = 0
    n = int(tokens[idx]); idx += 1
    m = int(tokens[idx]); idx += 1

    matrix = []
    for _ in range(n):
        row = []
        for _ in range(m):
            row.append(int(tokens[idx])); idx += 1
        matrix.append(row)

    out_lines = []
    for j in range(m):
        col = [str(matrix[i][j]) for i in range(n)]
        out_lines.append(" ".join(col))
    return "\n".join(out_lines) + "\n"
